src/app-bar.jsx

```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
```

Comment:
Well, not depending on DOM..., it's too much for the current implementation. But at least I will not rely on ReactDOM and will use the string refs but functions.